### PR TITLE
platform-api docker config initial stage

### PR DESCRIPTION
**Pull request by hulinpiao**
**The changes introduced in this Pull Request include:**

- [x] Creation of a Dockerfile to containerize the platform_api resources.
- [x] Configuration of the database connection using Docker networking.
- [x] Updates to related configurations and operational steps in the README.md.

### CODE REVIEW

File: Dockerfile
Comment (hulinpiao) [inline Dockerfile:15]: To whom review it：
it is initial docker config step, which simply run docker on localhost. so, I've only run 'yarn start' command for initial docker setup. However, we should consider the code build, test, format etc before the service running.

**Comment by hulinpiao** [comment]
@Chris-Wang , I found this "platform_api" repo can request up to 1 reviewer for PR. 
can we get more people to review the MR ? I would like to get review from other devops colleagues.

File: README.md
Comment (Chris-Wang) [inline README.md:78]: is this step essential for local development, if not, please consider marking it as `optional` with some comments

File: Dockerfile
Comment (Chris-Wang) [inline Dockerfile]: same as frontend, consider using `alpine`?

File: Dockerfile
Comment (Chris-Wang) [inline Dockerfile]: you can keep it aligned with your frontend docker file by removing `install`

File: tsconfig.json
Comment (Chris-Wang) [inline tsconfig.json:1]: please leave comments why some of the config and their order has been changed?

File: src/app.module.ts
Comment (Chris-Wang) [inline src/app.module.ts:1]: lack adding DB_HOST in env

**Review by Chris-Wang — CHANGES_REQUESTED** [review]

File: tsconfig.json
Comment (hulinpiao) [reply tsconfig.json:1]: I have sorted the JSON values/parameters in alphabetical order, as it is a good practice to organize parameters in configuration files. As a side note, the extension I use for sorting JSON and YAML files is called 'Sort Everything.'

File: Dockerfile
Comment (hulinpiao) [reply Dockerfile]: removed

File: Dockerfile
Comment (hulinpiao) [reply Dockerfile]: done.

File: src/app.module.ts
Comment (hulinpiao) [reply src/app.module.ts:1]: You're right, Chris. The update didn't push because .gitignore prevents .env from merging into the remote repository. I'll update the README.md file to include a note about adding DB_HOST

File: README.md
Comment (hulinpiao) [reply README.md:78]: It's not particularly useful for local development work. 
However, this step is vital for dockerizing the backend and connecting it to the DB container locally, a process that can also aid in preparing for cloud deployment through dockerization. 

Following your suggestion, I’ve marked it as optional and included some comments for clarification.

**Review by Chris-Wang — CHANGES_REQUESTED** [review]
Await Readme to be updated as discussed

**Review by Chris-Wang — CHANGES_REQUESTED** [review]
Need:
- pull latest dev to your branch, some code is duplicated
- resolve conflict
- squash commits after merge latest dev

**Comment by hulinpiao** [comment]
> Need:
> 
> * pull latest dev to your branch, some code is duplicated
> * resolve conflict
> * squash commits after merge latest dev

Thanks.  solved the duplicated.

**Review by Chris-Wang — APPROVED** [review]
LGTM

**Pull request merged**
